CHANGELOG.md

```diff
@@ -6,6 +6,32 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org).
```

Comment:
The release checklist in the PR description is out of date, did you want to update it, or open a new ticket with the latest steps?
https://github.com/ZcashFoundation/zebra/blob/main/.github/PULL_REQUEST_TEMPLATE/release-checklist.md#release-zebra

Reply:
Rebasing doesn't change the release checklist in this PR's description, which we are following to tag & test this release.
<img width="893" alt="Screenshot 2023-06-14 at 08 33 16" src="https://github.com/ZcashFoundation/zebra/assets/8951843/5693d340-055f-466f-bec7-a0469b26126e">


Would it help if I updated this PR's description?